### ISSUE TITLE
App packages launched from the VS IDE do not have dependent DLLs

### ISSUE THREAD

**Issue by sylveon** [microsoft/vcpkg#16184]
**Describe the bug**
When launching an app package from the VS IDE which refers to DLL from vcpkg, this error is generated because DLLs are not copied with the executable:
![image](https://user-images.githubusercontent.com/6440374/107730797-51659880-6cc2-11eb-8ba0-93a5a8a5bfd0.png)

**Environment**
- OS: Windows 10 19042
- Compiler: Microsoft Visual Studio Community 2019 Preview Version 16.9.0 Preview 4.0

**To Reproduce**
1. vcpkg integrate install
2. vcpkg install fmt
3. Create a new Windows Application project from Visual Studio (C++ > Windows > Desktop > Windows Desktop Application)
4. In the main cpp file, add `#include <fmt/format.h>`
5. Replace `// TODO: Place code here.` by `fmt::print("test");`
6. Right-click the solution in the solution browser and click on Add > New Project
7. Create an app packaging project (C++ > Windows > Desktop > Windows Application Packaging Project).
8. In the solution explorer, right-click on Applications within the packaging project, select Add references and add the desktop app project.
9. Right-click the app project under `Applications` and click `Set as entry point`
10. Right-click the packaging project and click "Set as Startup project"
11. Hit F5 to launch the project

Solution which reproduces the error:
[WindowsProject7.zip](https://github.com/microsoft/vcpkg/files/5969712/WindowsProject7.zip)

**Expected behavior**
The DLLs are copied with the main EXE in `<packaging project>\bin\<platform>\<configuration>\AppX`

**Failure logs** 
N/A

**Additional context**
Caused by https://github.com/microsoft/vcpkg/commit/401b26c9866a071f645ae57831caa419b2c48be1#diff-705430a4414b12ed477c77d25c696237d3936f152c0c63a4c8e240bbeb5dc3bbR100

**Comment by sylveon** [comment]
The cause seems to be that when the app is packaged within the IDE, LinkSkippedExecution is true:
![image](https://user-images.githubusercontent.com/6440374/107731257-80303e80-6cc3-11eb-93e5-b2c5cd97ff1b.png)

**Comment by soumyamahunt** [comment]
@sylveon are you still facing this issue?? Also I am not sure about `LinkSkippedExecution` causing this issue as I am still facing this issue even after removing `'$(LinkSkippedExecution)' != 'true'`.

**Comment by sylveon** [comment]
Yes, I am still facing this issue. Removing `LinkSkippedExecution` stopped working in 16.10 Preview 1.

**Comment by soumyamahunt** [comment]
> Yes, I am still facing this issue. Removing `LinkSkippedExecution` stopped working in 16.10 Preview 1.

I didn't have to remove `LinkSkippedExecution` until I was using 16.9 now that I upgraded to 16.9.1 I am starting to face this issue. Seems more like a Visual Studio bug than vcpkg bug to me.

**Comment by sylveon** [comment]
Can someone please look at this issue? I tried diagnosing the issues on 16.10 myself with build logging but didn't manage to find anything conclusive :(

It's making development very painful because every time I do a clean build I have to manually fish out the DLLs i need and copy-paste them into the AppX directory

**Comment by soumyamahunt** [comment]
> development very painful because every time I do a clean build I have to

I think vcpkg team's plate is full now, you are better off creating issue in [Project Reunion](https://github.com/microsoft/ProjectReunion) repo just to get some extra hands on deck.

**Comment by sylveon** [comment]
This is still an issue with VS 2022.

**Comment by nlogozzo** [comment]
Can confirm...this is still a thing!

**Comment by sylveon** [comment]
I just gave up and avoid using dependencies which require DLLs. Not a good experience :/

**Comment by JonLiu1993** [comment]
@sylveon, Sorry reply you so late, I use Visual Studio 2019 on windows 10 machine following your steps reproduce this error, But I did not reproduce your error:
````
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\DesktopBridge\Microsoft.DesktopBridge.targets(16,3): error MSB4019: The imported project "C:\Program Files\dotnet\sdk\2.1.519\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.NuGet.targets" was not found. Confirm that the expression in the Import declaration "C:\Program Files\dotnet\sdk\2.1.519\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.NuGet.targets" is correct, and that the file exists on disk.
````
![image](https://user-images.githubusercontent.com/63675417/172978783-6b34af35-f1ba-4125-a869-3b913f4f5ae9.png)

![image](https://user-images.githubusercontent.com/63675417/172978693-d389976d-7208-497c-aadc-f4e182c99086.png)

![image](https://user-images.githubusercontent.com/63675417/172977815-9b87ce46-90e9-4012-b86e-2b6f0bb79990.png)

**Comment by sylveon** [comment]
Does launching the packaging project works? This is what is giving me issues. My app can't be launched without the packaging project because it relies on feature that require a package identity.

**Comment by JonLiu1993** [comment]
@sylveon, I reinstalled some missing components and finally successfully reproduced your problem by following your reproduction steps

![image](https://user-images.githubusercontent.com/63675417/172981847-fab019d7-9d68-4458-b981-dfe0dd5f7b53.png)

**Comment by 3UR** [comment]
sad to see this issue is still around after 4 years :/ currently having it happen to me any workarounds?
